include/boost/geometry/algorithms/detail/sub_range.hpp

```diff
@@ -44,7 +49,7 @@ struct sub_range<Geometry, Tag, false>
 template <typename Geometry>
 struct sub_range<Geometry, polygon_tag, false>
 {
-    typedef typename geometry::ring_type<Geometry>::type & return_type;
+    typedef typename geometry::ring_type<Geometry>::type const& return_type;
```

Comment:
Why is this a problem? Isn't `ring_type<>` returning a type of rings suitable for the passed Geometry? So const ring for const Geometry and non-const otherwise?